code/modules/power/powernets/local_powernet.dm

```diff
@@ -0,0 +1,188 @@
+/**
+  * # local_powernet
+  *
+  * Manages all power related mechanics for a single /area
+  * Machines in areas will directly register to this datum in order to recieve power
+  *
+  * Machine/Turf/Item -> Local Powernet -> APC -> Terminal -> Wirenet
+  *
+  * Non-machines cannot register to the local powernet and should instead listen for power change signals
+*/
+/datum/local_powernet
+	/// The area this local powernet is attached to
+	var/area/powernet_area = null
+	/// The APC associated with this powernet
+	var/obj/machinery/power/apc/powernet_apc = null
```

Comment:
Make sure to clear this ref in `apc/Destroy`. Or do signal shenanigans.

Reply:
I believed it's already handled in apc destroy